grails-app/views/returns/show.gsp

```diff
@@ -278,18 +278,15 @@
                         <tr class="prop">
                             <td class="name">
                                 <g:message code="stockMovement.dateShipped.label"/>
-                                <g:if test="${stockMovement?.shipment?.status?.code==org.pih.warehouse.shipping.ShipmentStatusCode.PENDING}">
-                                    <small><g:message code="default.expected.label"/></small>
-                                </g:if>
                             </td>
                             <td class="value">
-                                <g:if test="${stockMovement?.dateShipped}">
+                                <g:if test="${stockMovement?.shipment?.hasShipped()}">
```

Comment:
is this changing to a boolean? That's what we want to do? Same question for the stock movement step.

Reply:
hasShipped() method that is called, is a method on Shipment domain, which returned type is Boolean (it checks if any event on the shipment contains an event with the SHIPPED eventTypeCode)

Reply:
Before, we were supposed to display the Date shipped even for an SM that was not shipped, this is why before we were checking the dateShipped (even though the dateShipped in this case might be misleading, because it’s not in fact the dateShipped, but the expectedDeliveryDate if I remember correctly).
Now we want to display this date with the user, only if an SM is really shipped.

Reply:
that makes sense. Thanks for the explanation!